bot/exts/fun/user_events.py

```diff
@@ -0,0 +1,603 @@
+
+    def cog_unload(self) -> None:
+        """Cancel scheduled tasks."""
+        self.scheduler.cancel_all()
+
+    async def restart_event_reminders(self) -> None:
+        """Restart scheduled event reminders when bot restarts."""
+        await self.bot.wait_until_guild_available()
+        scheduled_events = await self.bot.api_client.get("bot/scheduled-events")
+
+        for event in scheduled_events:
+            # If event is live
+            if datetime.now() > parse(event["start_time"]).replace(tzinfo=None):
+                self.schedule_event_end_reminder(event)
+
+            else:
+                await self.schedule_event_start_reminder(event)
+
+    @staticmethod
+    def not_scheduled() -> str:
+        """To indicate user event is not scheduled."""
+        return "Not scheduled"
+
+    @staticmethod
+    def live() -> str:
+        """To indicate user event is live."""
+        return "Live"
+
+    @staticmethod
+    def scheduled(start_datetime: datetime, end_datetime: datetime) -> str:
+        """
+        To indicate user event is scheduled.
+
+        The following code returns(an example):
+
+        Scheduled on 7th November 2020,
+        from 17:30 UTC To 18:00 UTC
+        (24-hour format).
+        """
+        readable_date = (
+            f"{start_datetime.day}{DATE_PREFIX.get(start_datetime.day, 'th')} "
+            f"{list(calendar.month_name)[start_datetime.month]} {start_datetime.year}"
+        )
+        readable_time = (
+            f"from {start_datetime.time().strftime('%H:%M')} UTC "
+            f"To {end_datetime.time().strftime('%H:%M')} UTC\n(24-hour format)."
+        )
+        status = f"Scheduled on {readable_date},\n{readable_time}"
+        return status
+
+    @property
+    def developers_role(self) -> Role:
+        """Return guild developers role."""
+        return self.guild.get_role(Roles.verified)
+
+    @property
+    def user_event_ongoing_role(self) -> Role:
+        """Return guild user-event-ongoing role."""
+        return self.guild.get_role(Roles.user_event_ongoing)
+
+    @property
+    def guild(self) -> Guild:
+        """Return guild instance."""
+        return self.bot.get_guild(Server.id)
+
+    @property
+    def user_event_coord_channel(self) -> TextChannel:
+        """Return #user-events-coordinators channel."""
+        return self.bot.get_channel(Channels.user_event_coordinators)
+
+    @property
+    def user_event_announcement_channel(self) -> TextChannel:
+        """Return #user-events-announcement channel."""
+        return self.bot.get_channel(Channels.user_event_announcements)
+
+    @property
+    def user_events_list_channel(self) -> TextChannel:
+        """Return #user-events-list channel."""
+        return self.bot.get_channel(Channels.user_event_list)
+
+    @property
+    def user_event_voice_channel(self) -> VoiceChannel:
+        """Return #user-events-voice channel."""
+        return self.bot.get_channel(Channels.user_event_voice)
```

Comment:
I don't know 100% how discord.py caching works, but I think there might be cases when these don't exist in cache.

Reply:
These commands are rarely used tho, it shouldn't matter much.
If incase its a concern, I guess I could declare all the roles and channels in the `__init__()` method, what do you say?

Reply:
I'd say more `self.bot.get_xxxx(yyy) or await self.bot.fetch_xxxx()`, but this mean that you have to make put them to async function. I think yes, you should create them inside task that is created in `__init__` that adds them to variables, but use this `or`.

Reply:
At least when you don't change it, make return type `Optional` and `None` check in code.

Reply:
I like the idea of loading them inside a task created in __init__ method.